src/NHibernate/Async/Transaction/AdoNetTransactionFactory.cs

```diff
@@ -40,35 +40,25 @@ public virtual Task ExecuteWorkInIsolationAsync(ISessionImplementor session, IIs
 			async Task InternalExecuteWorkInIsolationAsync()
 			{
 
-				DbConnection connection = null;
 				DbTransaction trans = null;
-				// bool wasAutoCommit = false;
+				// We make an exception for SQLite and use the session's connection,
+				// since SQLite only allows one connection to the database.
+				var connection = session.Factory.Dialect is SQLiteDialect
+				? session.Connection
```

Comment:
@maca88, small indentation glitch here with async generator, that you may be interested to look into. Nothing important, it is just for polishing it if you wish.

Reply:
Fixed in #2363.